File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlArithmeticBinaryExpr.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlArithmeticBinaryExpr implements ISqlExpression {
    SqlType: SqlType = SqlType.ArithmeticBinaryExpr;
    Span: TextSpan = new TextSpan();
    Left!: ISqlExpression;
    Right!: ISqlExpression;
    Operator: string = '';

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_SqlArithmeticBinaryExpr(this);
    }

    ToSql(): string {
        return `${this.Left.ToSql()} ${this.Operator} ${this.Right.ToSql()}`;
    }
} 

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlArithmeticBinaryExpr.cs
namespace T1.SqlSharp.Expressions;

public class SqlArithmeticBinaryExpr : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.ArithmeticBinaryExpr;
    public TextSpan Span { get; set; } = new();

    public required ISqlExpression Left { get; set; }
    public ArithmeticOperator Operator { get; set; }
    public required ISqlExpression Right { get; set; }

    public string ToSql()
    {
        return $"{Left.ToSql()} {Operator.ToSql()} {Right.ToSql()}";
    }
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_ArithmeticBinaryExpr(this);
    }
}